File: MyAnimeListJava/src/main/java/com/example/myanimelistjava/managers/SceneManager.java
package com.example.myanimelistjava.managers;

import com.example.myanimelistjava.MyAnimeListApplication;
import com.example.myanimelistjava.configurations.ViewConfig;
import javafx.application.Platform;
import javafx.fxml.FXMLLoader;
import javafx.scene.Scene;
import javafx.stage.Stage;
import javafx.stage.StageStyle;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.io.IOException;
import java.util.Objects;

import static com.example.myanimelistjava.views.Views.MAIN;
public class SceneManager {
    private static SceneManager instance;
    private final Class<?> appClass;
    Logger logger = LogManager.getLogger(SceneManager.class);

    private Stage mainStage;

    private SceneManager(Class<?> appClass) {
        this.appClass = appClass;
        logger.info("SceneManager created");
    }

    public static SceneManager getInstance(Class<?> appClass) {
        if (instance == null) {
            instance = new SceneManager(appClass);
        }
        return instance;
    }

    public void initSplash(Stage stage) throws IOException, InterruptedException {
        Platform.setImplicitExit(false);
        logger.info("Iniciando Splash");
        FXMLLoader fxmlLoader = new FXMLLoader(MyAnimeListApplication.class.getResource("views/splash-view.fxml"));
        Scene scene = new Scene(fxmlLoader.load(),Double.parseDouble(ViewConfig.WIDTH.getValue()), Double.parseDouble(ViewConfig.HEIGHT.getValue()));
        stage.setTitle("Programa");
        stage.setResizable(false);
        stage.setScene(scene);
        stage.initStyle(StageStyle.TRANSPARENT);
        stage.show();
    }

    public static SceneManager get() {
        return instance;
    }

    public void initMain() throws IOException {
        logger.info("Iniciando Main");
        Platform.setImplicitExit(true);
        FXMLLoader fxmlLoader = new FXMLLoader(Objects.requireNonNull(appClass.getResource("views/inicioSesion-view.fxml")));
        Scene scene = new Scene(fxmlLoader.load(), 600.0, 450.0);
        Stage stage = new Stage();
        stage.setResizable(false);
        stage.setTitle("Log In");
        logger.info("Scene loaded");
        stage.setScene(scene);
        mainStage = stage;
        stage.show();
    }
}

File: MyAnimeList/src/main/kotlin/com/example/myanimelist/managers/SceneManager.kt
package com.example.myanimelist.managers

import com.example.myanimelist.extensions.loadScene
import com.example.myanimelist.utils.*
import javafx.application.Platform
import javafx.fxml.FXMLLoader
import javafx.scene.Node
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.scene.image.Image
import javafx.stage.Modality
import javafx.stage.Stage
import javafx.stage.StageStyle
import java.io.IOException

object SceneManager {
    lateinit var appClass: Class<*>
    private var mainStage: Stage? = null

    inline fun <reified T> setAppClass() {
        this.appClass = T::class.java
    }

    //Si se usa mas de una vez dejar openStageAbout aqui o en un utils
    fun openStageAbout() =
        Stage().loadScene(ABOUT, ABOUT_WIDTH, ABOUT_HEIGHT) {
            title = "About"
            isResizable = false
            initModality(Modality.APPLICATION_MODAL)
            addIconStage(this)
        }.show()

    @Throws(IOException::class, InterruptedException::class)
    fun initSplash(stage: Stage) {
        Platform.setImplicitExit(false)
        println("Iniciando Splash")
        stage.loadScene(SPLASH) {
            addIconStage(stage)
            stage.title = "Splash"
            stage.isResizable = false
            stage.initStyle(StageStyle.TRANSPARENT)
        }.show()
    }


    @Throws(IOException::class)
    fun initMain() {
        Platform.setImplicitExit(true)
        mainStage = Stage().loadScene(LOGIN, WIDTH, HEIGHT) {
            isResizable = false
            title = "Login"
            initStyle(StageStyle.DECORATED)
            addIconStage(this)
        }
        println("Inicio listo")
        mainStage?.show()
    }


    @Throws(IOException::class)
    fun changeScene(node: Node, view: String) {
        println("Loading scene ")
        val stage = node.scene.window as Stage
        //oldStage.hide(); // Oculto la anterior
        val root = FXMLLoader.load<Parent>(
            appClass.getResource(view) ?: throw Exception("")

        )
        val newScene = Scene(root, 600.0, 450.0)
        println("Scene loaded")
        stage.scene = newScene
        stage.show()
    }


    /**
     * Para añadir el icono al crear stage más rápido
     * @param stage escena a añadir el icono
     */
    fun addIconStage(stage: Stage) = stage.icons.add(Image(ResourcesManager.getIconOf("icono.png")))

}